src/cdk-experimental/listbox/listbox.ts

```diff
@@ -143,13 +205,69 @@ export class CdkListbox implements AfterContentInit, OnDestroy {
     this._disabled = coerceBooleanProperty(value);
   }
 
+  /** Whether the listbox will use active descendant or will move focus onto the options. */
+  @Input()
+  get useActiveDescendant(): boolean {
+    return this._useActiveDescendant;
+  }
+  set useActiveDescendant(shouldUseActiveDescendant: boolean) {
+    this._useActiveDescendant = coerceBooleanProperty(shouldUseActiveDescendant);
+  }
+
+  ngOnInit() {
+    this._selectionModel = new SelectionModel<CdkOption>(this.multiple);
+  }
+
   ngAfterContentInit() {
-    this._listKeyManager = new ActiveDescendantKeyManager(this._options)
-      .withWrap().withVerticalOrientation().withTypeAhead();
+    this._initKeyManager();
+    this._initSelectionChange();
+    this._initSelectionModel();
   }
 
   ngOnDestroy() {
     this._listKeyManager.change.complete();
+    this._destroyed.next();
+    this._destroyed.complete();
+  }
+
+  private _initKeyManager() {
+    this._listKeyManager = new ActiveDescendantKeyManager(this._options)
+        .withWrap().withVerticalOrientation().withTypeAhead();
+
+    this._listKeyManager.change.pipe(takeUntil(this._destroyed)).subscribe(() => {
+      this._updateActiveOption();
+    });
+  }
+
+  private _initSelectionChange() {
+    this.optionSelectionChanges = defer(() => {
```

Comment:
Does this need to be a function that gets called, or could `optionSelectionChanges` be set just as part of its definition?

Reply:
Yeah it can be part of the definition. I can also make it readonly then. I'll change.